Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Systems/ArtifactInteractionTriggerSystem.cs

```diff
@@ -0,0 +1,34 @@
+using Content.Server.Xenoarchaeology.XenoArtifacts.Triggers.Components;
+using Content.Shared.ActionBlocker;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Helpers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+
+namespace Content.Server.Xenoarchaeology.XenoArtifacts.Triggers.Systems;
+
+public class ArtifactInteractionTriggerSystem : EntitySystem
+{
+    [Dependency] private readonly ArtifactSystem _artifactSystem = default!;
+    [Dependency] private readonly ActionBlockerSystem _actionBlocker = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ArtifactInteractionTriggerComponent, InteractHandEvent>(OnInteract);
+    }
+
+    private void OnInteract(EntityUid uid, ArtifactInteractionTriggerComponent component, InteractHandEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        // standard interactions check
+        if (!args.InRangeUnobstructed())
+            return;
+        if (!_actionBlocker.CanInteract(args.User) || !_actionBlocker.CanUse(args.User))
+            return;
```

Comment:
do these really need to be tested if `InteractHandEvent` is successfully being raised? @ElectroJr

Reply:
`_actionBlocker` used in more than 50 instance and in all my previous PRs. So I guess, it better to do this checks.

Reply:
First line of InteractHand runs CanInteract, and then InteractionActivate is run as the fallback from InteractHand so I don't think this is needed.

Reply:
I delete `_actionBlocker`. Tested with handcuffed player, seems to work fine without check.